lua/fspectate/sv_init.lua

```diff
@@ -241,3 +242,42 @@ local function fixAdminModIncompat()
 end
 
 hook.Add( "InitPostEntity", "fSpectate", fixAdminModIncompat )
+
+local lastChips = ""
+timer.Create( "fSpectateCheckChips", 0.5, 0, function()
```

Comment:
I don't love this, but it's probably fine.

I might try to keep an in memory list of starfalls and e2s as they're created and destroyed, as that's probably the most expensive part of this process.

If you opt not to do that, you could at least do a manual loop of `ents.GetAll` - I believe it's more efficient than doing two `findByClass`es

Reply:
iirc findbyclass is C++ and much more efficient than ents.getall

Reply:
> I might try to keep an in memory list of starfalls and e2s as they're created and destroyed
I wanted to do that but considering chips can change their name i felt like this was the better alternative.